e2e/pages/RegistrationDetails/RegistrationDetailsPage.ts

```diff
@@ -238,6 +238,54 @@ class RegistrationDetails {
     ).toContain(userName);
     expect(await timeStampLocator.textContent()).toContain(date);
   }
+
+  async addNote(actions: string, addNote: string) {
+    await this.page.getByText(actions).click();
+    await this.page.getByText(addNote).click();
+  }
+
+  async writeNote(placeholder: string, note: string, buttonName: string) {
+    const okButton = this.page.getByRole('button', {
+      name: buttonName,
+    });
+    await this.page.getByPlaceholder(placeholder).fill(note);
+    await this.page.waitForLoadState('networkidle');
+    await expect(okButton).toBeEnabled();
+
+    for (let i = 0; i < 2; i++) {
+      await okButton.waitFor({ state: 'visible' });
+      await okButton.click();
+    }
+  }
+
+  async validateNewNoteTile(
+    changeTitle: string,
+    userName: string,
+    date: string,
+    noteContent: string,
+  ) {
+    const historyTile = this.historyTile.nth(0);
```

Comment:
Maybe this could be slightly more generic at little extra cost:

```suggestion
  async validateNoteTile(
    noteIndex: number,
    changeTitle: string,
    userName: string,
    date: string,
    noteContent: string,
  ) {
    const historyTile = this.historyTile.nth(noteIndex);
```